File: signature/corejs.go
package signature

var CoreJsSignature = Signature{
	Name:        "core-js",
	Description: "Modular standard library for JavaScript.",
	Cpe:         "",

	DetectPattern: DetectPattern{
		Bodies: []string{
			"github.com/zloirock/core-js",
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			`https://github.com/zloirock/core-js/blob/v(\d+\.\d+\.\d+)/LICENSE`,
		},
	},
}

File: src/signatures/technologies/corejs.ts
import type { Signature } from "../_types.js";

export const corejsSignature: Signature = {
  name: "core-js",
  description:
    "core-js is a modular standard library for JavaScript, with polyfills for cutting-edge ECMAScript features.",
  rule: {
    confidence: "high",
    javascriptVariables: {
      "__core-js_shared__": "",
      "__core-js_shared__.versions.0.version": "(\\d+\\.\\d+\\.\\d+)",
      "_babelPolyfill": "",
      "core": "",
      "core.version": "(\\d+\\.\\d+\\.\\d+)",
    },
  },
};

